lib.nix

```diff
@@ -96,6 +96,26 @@ let
         [ pythonPackages.${drvAttr} or (throw "unsupported build system ${buildSystem}") ]
       );
 
+  # Provides a source filtering mechanism that tries to do the right thing as much as possible:
+  # - Filters gitignore (if it exists, non-recursive)
+  # - Filters pycache/pyc files
+  # - Uses cleanSourceFilter to filter out .git/.hg, .o/.so, editor backup files & nix result symlinks
+  cleanPythonSources = src: let
```

Comment:
In general this all looks good to me, I just first want to try this on some project(s) to see if everything works out as expected. Beyond that:

Filtering on `.gitignore` makes perfect sense and I can't think of any reason not to do that. As for `__pycache__` and `.pyc` files I wonder if there is any chance that a package might actually need those? What if such files are included for testing? (Just like this PR adds a `.pyc` file).

One answer to that is _"Well then you can just specify `src` if you don't want filtering"_. This is actually a bit odd. Filtering happens as an implicit consequence of specifying `src`:

```nix
# this will be filtered
mkPoetryApplication {
 projectDir = ./.;
}
```
```nix
# this will not be filtered
mkPoetryApplication {
 projectDir = ./.;
 src = ./.;
}
```

I think this needs to be more explicit. Not sure about the best approach from the top of my head though. No filtering by default but there is a `poetry2nix.filterSrc` ? Add a `doFilter ? true` attribute? Not sure, what do you think?

Reply:
You should never have to distribute the `.pyc` files or `__pycache__` files. They are just an interpretation of the code that removes a parsing step from the interpreter.

I think when you supply `src` you should still have full control of what gets passed in without some magic filtering happening (by default).

Reply:
While I agree the current usage is a bit odd I'm not sure what the API would look like otherwise.
I agree with @andir that I expect `src` to be passed without magic filtering if passed explicitly.

Reply:
I've updated the docs. Maybe just documenting the behaviour is sufficient?